src/hooks/useConnectionForm.ts

```diff
@@ -362,12 +362,16 @@ export const useConnectionForm = (validationSchema: ZodObject<ZodRawShape>, mode
 		}
 	};
 
-	const handleCustomOauth = async (oauthConnectionId: string, integrationName: string) => {
+	const handleCustomOauth = async (
+		oauthConnectionId: string,
+		integrationName: keyof typeof Integrations | typeof defaultGoogleConnectionName,
+		authType?: ConnectionAuthType.CustomOAuth | ConnectionAuthType.Oauth
+	) => {
 		setIsLoading(true);
 		try {
 			await VariablesService.setByConnectiontId(oauthConnectionId, {
 				name: "auth_type",
-				value: ConnectionAuthType.Oauth,
+				value: authType || ConnectionAuthType.Oauth,
```

Comment:
https://github.com/autokitteh/web-platform/pull/936/files#r1935390033

```suggestion
				value: authType,
```

Reply:
changed